### ISSUE TITLE
Error:  Exception: Class 'quizaccess_proctoring\quiz_settings' not found 

### ISSUE THREAD

**Issue by wstemb** [eLearning-BS23/moodle-quizaccess_proctoring#30]
After the deinstalation of the plugin, and installing it again,  pressing the button: "View proctoring report" I got the error 
" Exception: Class 'quizaccess_proctoring\quiz_settings' not found "

Before it was working well.  

Moodle 3.9.2. 
Plugin Installed version 1.1.1 2020091012

Walter

**Comment by wstemb** [comment]
Additional comments and info on my comment 

If I enable the Safe Exam Browser in the Course where I activated the Proctoring plugin and I enter the quiz as Teacher or Manager, the Button "View proctoring report" is working well, showing also the previously captured shots (so the capturing component of the plugin during the exams is OK and the shots are in the system, only the "View" function is in error. 

Unfortunately, I do not know how to enable the camera in SEB, so it is not a way to use both plugins together at my level of knowledge. 

A temporary workaround form me, at least during the tests can be to use the proctoring plugin without SEB for students during the exam, and to enable the SEB after the exam ta access the saved images. But it is not a right way to give the Proctoring plugin to other teachers.  

The reason for de-installing and installing again the plugin was that the Settings form (capture interval and capture size) is shown only once, on the first install. I did not find the way to reenter and change the data, since in the Plugin directory there is not a "Settings" link near the plugin, only Deinstall. 

Walter

**Comment by wstemb** [comment]
The error is triggered at the line 124 of the classes/helper.php, but I do not why it is working as is in a fresh install.

**Comment by wstemb** [comment]
When the error is present, it is triggered by the $OUTPUT->header(); instruction in report.php.  
During the execution, config.php is called, where is: $config = \quizaccess_proctoring\helper::get_proctoring_config_content($cmid);

get_proctoring_config_content() is defined in helper.php (line124), where quiz_settongs is called, and this is raising the exeption.

**Comment by AhnafMuttaki** [comment]
# Reinstalling the plugin shows settings not found error:
We have tested it in our system and found it to be working fine. Can you please test if only your moodle instance is facing this issue or not.

# Settings can be found while installation only:
You can still change the settings after installation. Please navigate to 
Site Administrator -> Go to Plugin Tab -> Under "Activity modules" you will find a link "Proctoring".
You can change the settings from there. No need to reinstall the plugin.

# Your issue regarding safe exam browser is not clear. Can you please share some screen shot

**Comment by wstemb** [comment]
As I wrote in previous posts, the error message on my moodle instance is triggered when "config.php" is called during the $OUTPUT->header(); in report.php. In this case, config.php is calling  classes/helper.php, where the **quiz_settings** class is mentioned and this is triggering the error. This is happening only when the "Edit on" mode is activated. 

On another instance, where I did not reinstall the plugin, my impression is the config.php is never called (i tried to echo some text to web page). So the error (which is still present in code, another my impression) is never triggered.

I will try to reproduce the scenario on this instance.

SEB is not important any more. I just found during the tests that if I worked as a teacher in SEB the error was not present. Since I found the workaround (Turn editing off), I can use the plugin on my moodle. I just placed an echo in the config.php with the instruction  to users to try "Turn Edit Off" to avoid the error.
